pkg/datapath/iptables/iptables.go

```diff
@@ -695,26 +677,99 @@ func (m *IptablesManager) installStaticProxyRules() error {
 	return err
 }
 
+var tproxyMatch = regexp.MustCompile("CILIUM_PRE_mangle .*cilium: TPROXY")
+
+// copies old proxy rules
+func (m *IptablesManager) copyProxyRules(oldChain string, match string) {
+	copyRules := func(prog string, table string, re *regexp.Regexp, match, oldChain, newChain string) {
+		output, err := runProgCombinedOutput(prog, append(m.waitArgs, "-t", table, "-S"), true)
+		if err != nil {
+			log.WithError(err).Infof("Can't list %s %s table rules", prog, table)
+			return
+		}
+		scanner := bufio.NewScanner(bytes.NewReader(output))
+		for scanner.Scan() {
+			rule := scanner.Text()
+			if re.MatchString(rule) && strings.Contains(rule, match) {
+				log.WithField(logfields.Object, logfields.Repr(rule)).Debugf("Considering copying %s TPROXY rule from %s to %s", prog, oldChain, newChain)
+				args, err := shellwords.Parse(strings.Replace(rule, oldChain, newChain, 1))
+				if err != nil {
+					log.WithError(err).WithField(logfields.Object, rule).Warnf("Unable to parse %s TPROXY rule", prog)
```

Comment:
Ditto here, I think we should log the consequence (potential impact on traffic selected by L7 rules). Presumably same below too.